core/src/storage2/collections/vec/mod.rs

```diff
@@ -303,4 +303,36 @@ where
         *self.len = last_index;
         Some(())
     }
+
+    /// Sets the elements at the given index to the new value.
+    ///
+    /// Won't return the old element back to the caller.
+    /// Prefer this operation over other method of overriding an element
+    /// in the storage vector since this is more efficient.
+    ///
+    /// # Panics
+    ///
+    /// If the index is out of bounds.
+    pub fn set(&mut self, index: u32, new_value: T) {
+        if let None = self.within_bounds(index) {
+            panic!(
+                "index out of bounds: the len is {} but the index is {}",
+                self.len(),
+                index
+            );
+        }
+
+        self.elems.put(index, Some(new_value));
+    }
+
+    /// Removes all elements from this vector.
+    pub fn clear(&mut self) {
+        if self.is_empty() {
+            return
+        }
+        for index in 0..self.len() {
+            self.elems.put(index, None);
+        }
+        *self.len = 0;
```

Comment:
Here we could make use of the `storage::Lazy::set` method if we just had it to further decrease the overhead. :upside_down_face: